Source/WebKit/Platform/cocoa/WebPrivacyHelpers.mm

```diff
@@ -406,6 +406,33 @@ static RestrictedOpenerType restrictedOpenerType(WPRestrictedOpenerType type)
     return it == m_restrictedOpenerTypes.end() ? RestrictedOpenerType::Unrestricted : it->value;
 }
 
+ResourceMonitorURLsController& ResourceMonitorURLsController::singleton()
+{
+    static MainThreadNeverDestroyed<ResourceMonitorURLsController> sharedInstance;
+    return sharedInstance.get();
+}
+
+void ResourceMonitorURLsController::prepare(CompletionHandler<void(WKContentRuleList*, bool)>&& completionHandler)
+{
+    ASSERT(RunLoop::isMain());
+    if (!PAL::isWebPrivacyFrameworkAvailable() || ![PAL::getWPResourcesClass() instancesRespondToSelector:@selector(prepareResouceMonitorRulesForStore:completionHandler:)]) {
+        completionHandler(nullptr, false);
+        return;
+    }
+
+    WKContentRuleListStore *store = [WKContentRuleListStore defaultStore];
+
+    [[PAL::getWPResourcesClass() sharedInstance] prepareResouceMonitorRulesForStore:store completionHandler:^(WKContentRuleList *list, bool updated, NSError *error) {
+        if (error) {
+            RELEASE_LOG_ERROR(ResourceLoadStatistics, "Failed to request resource monitor urls from WebPrivacy");
+            completionHandler(nullptr, false);
```

Comment:
You may want to follow the pattern in other classes in this file which stash the completionHandler into a static vector, and then `std::exchange` the callbacks once the result comes back. This effectively makes it so that multiple outgoing calls to `prepare` only result in one call to WebPrivacy.